Feature/sn auth package extraimprovements

NoelKova opened the pull request.

hashtagnulla committed:
feat: sn-auth package

hashtagnulla committed:
updated sn-auth-react with api calls and refactored token refreshing

hashtagnulla committed:
moving react dependency to peer dependencies

hashtagnulla committed:
moving react dependency to peer dependencies

hashtagnulla committed:
fix: refresh token

hashtagnulla committed:
raised version number

hashtagnulla committed:
bumped version number

hashtagnulla committed:
fix: blank screen on content actions

zoltanmarosvolgyi committed:
Fix UI elements and file download

zoltanmarosvolgyi committed:
Add ui fixe

zoltanmarosvolgyi committed:
Fix: ActionButtonWrapper CSS

zoltanmarosvolgyi committed:
admin ui extra improvements

zoltanmarosvolgyi committed:
admin ui extra imp. #2

zoltanmarosvolgyi committed:
add fixes

zoltanmarosvolgyi committed:
Fix import

zoltanmarosvolgyi committed:
Remove junks

zoltanmarosvolgyi committed:
Fix old functions

zoltanmarosvolgyi committed:
fix odataparams

zoltanmarosvolgyi committed:
Add selectionService to Grid

zoltanmarosvolgyi committed:
Add pipe to gitignore

zoltanmarosvolgyi committed:
fixes

zoltanmarosvolgyi committed:
fix tree

zoltanmarosvolgyi committed:
fixed locked cell in the grid

zoltanmarosvolgyi committed:
fixes

zoltanmarosvolgyi committed:
fix height to 25px

zoltanmarosvolgyi committed:
Design improvements

VargaJoe committed:
add build time auth switch

NoelKova committed:
keler to snauth and eslint fixes

NoelKova committed:
remove console logs

NoelKova committed:
grid selection filtered only

NoelKova committed:
update deprecated varsions

NoelKova committed:
update eslint-plugin-jsdoc

NoelKova committed:
downgrade eslint-plugin-jsdoc

NoelKova committed:
remove examples build from ci.yml

NoelKova committed:
put back examples build

NoelKova committed:
yml edits

NoelKova committed:
comment test

NoelKova committed:
revert yml changes

NoelKova committed:
copy ymls from develop

NoelKova committed:
Merge pull request #1641 from SenseNet/feature/sn-client-refactor

Sn Client Refactor

NoelKova committed:
login with both auth type at the same time

NoelKova committed:
Merge branch 'feature/sn-auth-package-extraimprovements-isswitch' into feature/sn-auth-package-extraimprovements

NoelKova committed:
Merge branch 'develop' into feature/sn-auth-package-extraimprovements

NoelKova committed:
add missing file

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/SenseNet/sn-client?pullRequest=1643) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/SenseNet/sn-client?pullRequest=1643) before we can accept your contribution.<br/>**2** out of **3** committers have signed the CLA.<br/><br/>:white_check_mark: NoelKova<br/>:white_check_mark: VargaJoe<br/>:x: Aux Ciprián<br/><hr/>**Aux Ciprián** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/SenseNet/sn-client?pullRequest=1643) it.</sub>

NoelKova committed:
remove storybook build

NoelKova committed:
remove storybook build

NoelKova committed:
ci on PR

NoelKova committed:
yml version

NoelKova committed:
remove ci on pull request

NoelKova committed:
remove exmaples build

NoelKova committed:
remove tests

NoelKova committed:
lint fixes

NoelKova committed:
minor lint fix

NoelKova commented:
After building to test, collect issues if any, then fix them up and after that we can merge into develop

NoelKova committed:
empty saved queries fix

NoelKova committed:
content explorer tree disabled fix

NoelKova committed:
copy dialog height fix

NoelKova committed:
edit view design

NoelKova committed:
edit view design 2

NoelKova committed:
reference picker

NoelKova committed:
copy move dialog

NoelKova committed:
reference picker height fix

NoelKova committed:
insert file into tinyMCE start

NoelKova committed:
Revert "insert file into tinyMCE start"

This reverts commit 7e8e48ef9f9abf2bf507ea9d142d8826c642f743.

NoelKova committed:
input updates

NoelKova committed:
auth redirect on logout fix

NoelKova committed:
multi auth type refactor

NoelKova committed:
repo login from url

NoelKova committed:
reference picker height

NoelKova committed:
show hide input descriptions

NoelKova committed:
show hide input descriptions 2

NoelKova committed:
tinyMCE file picker

NoelKova committed:
picker flex direction

NoelKova committed:
datepicker keyboard typing

NoelKova committed:
datepicker refactor

NoelKova committed:
settings error, grid extre scrollbar

NoelKova committed:
insert file path fix

NoelKova committed:
picker package fix

NoelKova committed:
tree remove cache when edit item

NoelKova committed:
add icons to public folder to fix CORS error

NoelKova committed:
icons fixes

NoelKova committed:
Revert "icons fixes"

This reverts commit 8624a91648e42f29b107cc283ba3dc666aed96d9.

NoelKova committed:
Revert "add icons to public folder to fix CORS error"

This reverts commit bf9a36ce1bab2909a65e9359991c806e9ed9b0ce.

NoelKova committed:
close disabled tree items

NoelKova committed:
tree resizer overlap

NoelKova committed:
grid header height fix

NoelKova committed:
icon for specific files

NoelKova committed:
excel icon change

NoelKova committed:
search results refator

NoelKova committed:
picker default path fix

VargaJoe committed:
feat(field-controls): show field titles as 'DisplayName (Name)' in al… (#1644)

* Field controls: Show field label as 'DisplayName (Name)' in all edit and browse UIs for consistency and clarity

NoelKova committed:
datepicker clear fix

NoelKova committed:
Merge branch 'feature/sn-auth-package-extraimprovements' of https://github.com/SenseNet/sn-client into feature/sn-auth-package-extraimprovements

NoelKova committed:
content explorer to root

NoelKova committed:
path fix

NoelKova committed:
content path fix

NoelKova committed:
reference picker refactor

NoelKova committed:
picker fixes

NoelKova committed:
pciker fix search selector

NoelKova committed:
icon in grid fix

NoelKova committed:
picker sort and not load fix

NoelKova committed:
search bar redesign

NoelKova committed:
search minor design fix

NoelKova committed:
searchbar debounce fix

NoelKova committed:
header link more visible + copy link to clipboard

NoelKova committed:
reference grid to ag grid

NoelKova committed:
settings and localisation open from tree

NoelKova committed:
modified by somebody fix

NoelKova committed:
header color changer for test cases can be set in portal settings

NoelKova committed:
remove log

NoelKova committed:
add new picker to tineyMCE editor

NoelKova committed:
textarea padding fix

NoelKova committed:
copy move new picker

NoelKova committed:
fix grid forever loading

NoelKova committed:
tooltips for grid

NoelKova committed:
trash column change

NoelKova committed:
grid actions button width fix

NoelKova committed:
delete grid sort by date

NoelKova committed:
webhooks picker change to new

NoelKova committed:
reference grid auto height fix

NoelKova committed:
edit view intezo mode for tree

NoelKova committed:
field boldness change

NoelKova committed:
swticher allignment

NoelKova committed:
save grid sorting

NoelKova committed:
fix copy picker

NoelKova committed:
home page info

NoelKova committed:
remove columns from grid when small screen

NoelKova committed:
minor bug fixes

NoelKova committed:
tree resizer pos fix

NoelKova committed:
picker and minor fixes

NoelKova committed:
custom label for highlighted fileds

NoelKova committed:
icon from path CORS error fix

NoelKova committed:
icon fixes v2

NoelKova committed:
icon fixes v3

NoelKova committed:
remove console logs

Commit:
local time fix

NoelKova committed:
Content button missing + minor changes

NoelKova committed:
remove logs

Commit:
Merge branch 'feature/sn-auth-package-extraimprovements' of https://github.com/SenseNet/sn-client into feature/sn-auth-package-extraimprovements

NoelKova committed:
add button error

NoelKova committed:
Merge branch 'feature/sn-auth-package-extraimprovements' of https://github.com/SenseNet/sn-client into feature/sn-auth-package-extraimprovements

NoelKova committed:
field descriptions between name and input

NoelKova committed:
selectionRootsFix

NoelKova committed:
same date format

NoelKova committed:
Audit Log Design

NoelKova committed:
audit log content history

NoelKova committed:
content history same fix

Commit:
user lock fix

NoelKova committed:
picker fix

NoelKova committed:
tul hosszu title leturja gombokat

NoelKova committed:
fix console errors

NoelKova committed:
picker fixes

NoelKova committed:
tree nav refactor

NoelKova committed:
permission fix, missing localization, avatar fixes

VargaJoe committed:
feat: add Docker containerization support (#1648)

* feat: add Docker containerization support

- Add Dockerfile for SenseNet client application
- Add docker-compose.dev.yml for development with hot reload
- Add docker-compose.prod.yml for production deployment
- Add .dockerignore for optimized builds
- Add .env.example for environment configuration

Usage:
- Development: docker-compose -f docker-compose.dev.yml up -d
- Production: docker-compose -f docker-compose.prod.yml up -d

* feat: add Docker CI/CD workflow

- Add GitHub Actions workflow for Docker image builds
- Builds and pushes to DockerHub with smart tagging
- Triggers on pushes to docker-containerization branch
- Triggers on PRs to develop, main, and sn-auth-package-extraimprovements
- Safely skips build if no Dockerfile exists
- Uses branch-based and SHA-based tags for versioning

* fix: Docker workflow tag format issue

- Fix invalid tag format causing build failure
- Use proper suffix/prefix for SHA tags
- Ensure Docker tag compliance

* fix: prevent duplicate workflow runs

- Skip draft PRs to avoid duplicate builds
- Only trigger PR builds when PR is ready for review
- Maintain push-based builds for active development

* remove space

* refactor: remove SHA-only Docker tags

- Remove standalone SHA tags to prevent DockerHub clutter
- Keep clean branch names and branch+SHA tags for useful scenarios
- Maintain manageable tag strategy for long-term development

* docs: update Docker documentation to reflect simplified setup

- Removed multi-stage Dockerfile complexity
- Updated documentation to match single Dockerfile approach
- Both dev and prod modes use 'yarn snapp start'
- Development uses volume mounts for hot reload
- Production uses built-in code without volumes
- Simplified and cleaner Docker setup

* fix: improve Docker setup and CI workflow

- fix CI workflow: push to DockerHub only on push events, not PRs
- add develop/main/feature branches to push trigger
- fix docker-compose.dev.yml: remove unsupported glob volume patterns
- fix docker-compose.prod.yml: use pre-built image, env_file, configurable port
- fix .gitignore: add .env and .env.*.local patterns

VargaJoe committed:
fix: allow external host headers when running in Docker

- webpack devServer allowedHosts controlled by ALLOWED_HOSTS env var
- docker-compose.dev.yml sets ALLOWED_HOSTS=all to allow nginx reverse proxy

VargaJoe committed:
fix: simplify allowedHosts config in webpack devServer

VargaJoe committed:
feat: pre-build app bundle in Dockerfile, serve static files

- add yarn snapp build to Dockerfile to compile bundle at image build time
- switch CMD from webpack-dev-server to 'serve' static file server
- container startup reduced from minutes to seconds
- reduce HEALTHCHECK start-period from 60s to 15s
- docker-compose.dev.yml still overrides CMD for local hot reload

VargaJoe committed:
fix: set NODE_ENV=production for production build in Dockerfile

marosvolgyiz committed:
feat: update docker-compose and babelrc configurations for improved development setup

marosvolgyiz committed:
feat: enhance form submission handling to prevent duplicate submits and improve async support

marosvolgyiz committed:
refactor: streamline Dockerfile and docker-compose for improved build process

marosvolgyiz committed:
feat: add CSV export functionality with dialog and localization support

marosvolgyiz committed:
feat: implement loading state management for grid and children components, enhance icon loading logic, and optimize CSV export handling

marosvolgyiz committed:
feat: add ZIP download functionality for selected items with success and error handling

marosvolgyiz committed:
feat: add FormPropertiesEditor component with localization support and JSON handling

marosvolgyiz committed:
feat: add AUIApplication content type and related documentation

- Introduced AUIApplication content type with custom HTML rendering capabilities.
- Added Admin UI Applications documentation for usage and API details.
- Implemented AUIApplicationView component to handle rendering of AUIApplication content.
- Updated Explore component to conditionally render AUIApplicationView.
- Enhanced IconFromPath component to accept repository as a parameter.
- Created example application for AUIApplication demonstrating content listing and editing.
- Added localization for content type templates.
- Improved zip download service with path sanitization.

marosvolgyiz committed:
AUIApplication

marosvolgyiz committed:
Add actionmenu to searchbar items

marosvolgyiz committed:
feat: enhance repo file plugin with download markup and URL normalization

marosvolgyiz committed:
feat: add HomeActivitySummary component and enhance Home with activity logging features

VargaJoe committed:
Merge/develop into sn auth extraimprovements (#1649)

* feature: schema selector field #2217

* feature: add extra class

* refactor: revert package manager

* remove package mamnager

* fix: undefined on no message

* refactor: modify comment

* change pre-comit hook

* refacotr builds

* revert  ci cahnges

* remove wake up notifyier

* remove story book ci

* fature: add link to refences

* fix: remove console log

* feature: loading state to operation

* merge: integrate develop into sn-auth-extraimprovements

- package.json: add packageManager and empty dependencies from develop
- operations dialog: combine null-safe elements?.map with InfluenceField support
- default-item-template: take develop version, export getAppPathAndContent/buildCustomPath helpers
- reference-grid: keep ag-grid table view, add clickable Path column navigation, accept paths/dialogTitleComponent props from develop

---------

Co-authored-by: Hassan Adam <hassana@sensenet.com>

marosvolgyiz committed:
Merge pull request #1650 from SenseNet/sn-auth-package-extraimprovements-refactor

Sn auth package extraimprovements refactor

VargaJoe committed:
fix docker image build

marosvolgyiz committed:
feat: enhance TypeScript configuration and webpack setup for improved path resolution and type checking

marosvolgyiz committed:
feat: enhance AUIApplication with route location handling and bridge location creation

marosvolgyiz committed:
feat: enhance sandbox attributes for AUIApplicationView component

marosvolgyiz committed:
feat: add theme support to AUIApplication and bridge components

VargaJoe committed:
usercolumn cell renderer array fix

VargaJoe committed:
Fix IMS user change password action

VargaJoe committed:
Fix command palette action context selection (#1651)

VargaJoe committed:
Merge branch 'develop' into feature/sn-auth-package-extraimprovements

VargaJoe committed:
feat(sensenet): add configurable folder-first sorting

marosvolgyiz committed:
refactor: simplify item activation logic in ContentList and Grid components

marosvolgyiz committed:
feat: enhance AUIApplicationBridge with binary response handling and add tests

marosvolgyiz committed:
feat: enhance FileUpload and EditView components with text binary handling and Monaco editor integration

VargaJoe committed:
feat(admin-ui): support multi-repository SNAuth sessions (#1652)

* feat(admin-ui): scope SNAuth sessions by repository

* feat(admin-ui): add SNAuth repository switcher

marosvolgyiz committed:
feat: enhance FileUpload component with text binary handling, editor resizing, and modified state indication

VargaJoe committed:
feat(admin-ui): improve mobile layout (#1653)

* feat(admin-ui): improve mobile layout

* fix(admin-ui): stabilize mobile dev view

* fix(admin-ui): tighten mobile tree spacing

* fix(admin-ui): improve mobile header controls

* fix(admin-ui): align mobile header actions

marosvolgyiz committed:
feat: implement favorites feature with UI components and backend integration

marosvolgyiz committed:
feat(upload): enhance upload functionality with detailed progress tracking and improved UI

- Introduced UploadAttemptDetails type to encapsulate upload metadata.
- Updated FileWithFullPath type to include upload attempt details.
- Refactored ProgressBar component for better status display and visual feedback.
- Enhanced UploadDialog with improved file handling, including skipped files and overall progress tracking.
- Added localization for new upload statuses and messages in both default and Hungarian languages.

marosvolgyiz committed:
feat: add image gallery functionality

- Implemented image gallery dialog to view images in a gallery format.
- Added `useImageGallery` hook for managing gallery state and opening the dialog.
- Integrated image gallery button in `ContentInfo` and context menu for image content.
- Enhanced image loading and error handling in the `ImageView` component.
- Created `ImageThumbnail` component for displaying image thumbnails.
- Updated localization for image gallery features in both default and Hungarian languages.
- Added utility functions for checking image content and fetching image URLs.
- Adjusted styles for image gallery dialog and grid components.

marosvolgyiz committed:
feat: implement column settings dialog with drag-and-drop functionality

- Added ColumnSettingsDialog component for managing column visibility and order.
- Integrated Material-UI components for better UI/UX.
- Implemented drag-and-drop for reordering columns.
- Added localization support for column settings.
- Created services for loading and saving column settings to the repository.
- Updated Grid component to support column settings dialog and apply legacy column settings.
- Introduced hooks for managing repository column settings.
- Enhanced error handling and loading states for column settings.

marosvolgyiz committed:
feat: add fullscreen edit functionality with UI integration and localization support

marosvolgyiz committed:
feat: enhance user menu with account settings and logout options, improve roles display with group names
